### DISCUSSION TITLE
Questions regarding XML deserialisation

### DISCUSSION THREAD

**Discussion by KDederichs** [symfony/symfony#47177] [Q&A]
Hi,

I'm having a problem where I need to deserialise an API response that's shaped like this:
```
<?xml version="1.0" encoding="ISO-8859-1"?>
<Daten>
<Result> <Code>00</Code> <Text>OK</Text>
</Result>
 <Artikellliste>
  <Artikel>
    SOME ATTRIBUTES HERE
  </Artikel>
  <Artikel>
    SOME ATTRIBUTES HERE
  </Artikel>
 </Artikellliste>
<Zusammenfassung>
<Anzahl>2</Anzahl> <Artikelsätze>1</Artikelsätze> <Ausführungen>2</Ausführungen> <LastIndex>2</LastIndex>
</Zusammenfassung>
```

and my DTO class looks like this:
```
class Daten
{
    private Result $Result;
    /** @var array<Artikel> */
    private array $Artikellliste;
    private Zusammenfassung $Zusammenfassung;
.
.
.
}
```

Now I'm running into two issues:

1) The Artikel attriube isn't recognized and thus skipped, what's the correct way of deserialising a structure like that?
2) The probably bigger problem: The Umlauts in the attributes in Zusammenfassung are causing deserialisation errors.
Now the question here is if that's something PHP does or if it's something that can be fixed in Symfony...

Anyone having dealt with similar stuff/ knows how to handle those?

**Comment by KDederichs** [comment]
Ok I answered 1. myself, you need to add a DTO for that, that then contains the Artikle (I'll add a PR for the docs with an example for that I think)

Still leaves me with 2. though if someone knows that one

**Comment by lyrixx** [comment]
We'll need more core to help you. I think if you provider a reproducer, it'll be easier to get an answer

**Comment by KDederichs** [comment]
Sure here you go:
https://github.com/KDederichs/xml_parse_reproducer

I noticed in the mean time if you replace their ISO encoding with UTF-8 it tends to work so I also made an example with that.

/test-bad and /test-good respectively

Example data is taken of of their docs.
